arcade/examples/minimap_camera.py

```diff
@@ -0,0 +1,212 @@
+        self.camera_sprites.use()
+
+        # Draw all the sprites.
+        self.wall_list.draw()
+        self.player_list.draw()
+
+        # Draw new minimap using the camera
+        self.camera_minimap.use()
+        self.clear(MINIMAP_BACKGROUND_COLOR)
+        self.wall_list.draw()
+        self.player_list.draw()
+
+        # Select the (unscrolled) camera for our GUI
+        self.camera_gui.use()
+
+        # Draw the GUI
+        arcade.draw_rectangle_filled(self.width // 2, 20, self.width, 40, arcade.color.ALMOND)
+        self.instructions.draw()
+
+    def on_key_press(self, key, modifiers):
+        """Called whenever a key is pressed. """
+
+        if key == arcade.key.UP:
+            self.player_sprite.change_y = PLAYER_MOVEMENT_SPEED
+        elif key == arcade.key.DOWN:
+            self.player_sprite.change_y = -PLAYER_MOVEMENT_SPEED
+        elif key == arcade.key.LEFT:
+            self.player_sprite.change_x = -PLAYER_MOVEMENT_SPEED
+        elif key == arcade.key.RIGHT:
+            self.player_sprite.change_x = PLAYER_MOVEMENT_SPEED
+        elif key == arcade.key.A:
+            self.selected_camera = self.camera_minimap
+        elif key == arcade.key.B:
+            self.selected_camera = self.camera_sprites
+        elif key == arcade.key.W:
+            self.selected_camera.zoom += 0.1
+        elif key == arcade.key.S:
+            if (self.selected_camera.zoom - 0.1) > 0:
+                self.selected_camera.zoom -= 0.1
+        elif key == arcade.key.I:
+            l, b, w, h = self.camera_minimap.viewport
+            self.camera_minimap.viewport = l + 100, b + 100, w - 100, h - 100
+        elif key == arcade.key.K:
+            l, b, w, h = self.camera_minimap.viewport
+            self.camera_minimap.viewport = l - 100, b - 100, w + 100, h + 100
+
+    def on_key_release(self, key, modifiers):
+        """Called when the user releases a key. """
+
+        if key == arcade.key.UP or key == arcade.key.DOWN:
+            self.player_sprite.change_y = 0
+        elif key == arcade.key.LEFT or key == arcade.key.RIGHT:
+            self.player_sprite.change_x = 0
+
+    def on_update(self, delta_time):
+        """ Movement and game logic """
+
+        # Call update on all sprites (The sprites don't do much in this
+        # example though.)
+        self.physics_engine.update()
+
+        # Center the screen to the player
+        self.camera_sprites.center(self.player_sprite.position, CAMERA_SPEED)
+
+        # Center the minimap viewport to the player in the minimap
+        self.camera_minimap.center(self.player_sprite.position, CAMERA_SPEED)
+        # print(self.camera_sprites.position, self.camera_sprites.get_adjusted_position(),
+        #       self.camera_minimap.position, self.camera_minimap.get_adjusted_position())
+        # print(self.camera_sprites.sync(self.camera_minimap))
+        print(self.camera_minimap.scale)
+        print(self.camera_minimap.projection)
```

Comment:
If I read the test output correctly, the second error is due to a recently added check to ensure examples do not print to the console. Try replacing these print statements with Text objects that are displayed inside the window.

Reply:
I’ll change this!

Reply:
Thanks!